### PR TITLE
Migrate from MonoTouch to Xamarin.iOS/Xamarin.Android unified API.

### PR DESCRIPTION
**Pull request by lrdcasimir**

### CODE REVIEW

**Comment by lrdcasimir** [comment]
This should allow the Xamarin SDK to be used in new versions of Xamarin.

File: Src/packages/repositories.config [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <repositories>
-  <repository path="..\packages.Buddy.Net45.config" />
-  <repository path="..\packages.Buddy.Windows.config" />
-  <repository path="..\packages.Buddy.WP8.config" />
-  <repository path="..\packages.Buddy.WP81.config" />
+  <repository path="../packages.Buddy.Android.config" />
```
Comment (shawnburke) [inline Src/packages/repositories.config]: does this work on windows too?

File: Src/PlatformAccess/IosPlatformAccess.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using System.Net;
```
Comment (shawnburke) [inline Src/PlatformAccess/IosPlatformAccess.cs]: looks great.  does android need to be updated too?

Comment (lrdcasimir) [inline Src/PlatformAccess/IosPlatformAccess.cs]: Updating the nuget config got android building alongside this with the Newtonsoft PCL. I'll test that we're on the right target framework there.

File: Src/packages/repositories.config (hunk 1 above)
Comment (lrdcasimir) [inline Src/packages/repositories.config]: Good catch, looks like this needs some tweaking.

File: Src/PlatformAccess/IosPlatformAccess.cs (hunk 2 above)
Comment (lrdcasimir) [inline Src/PlatformAccess/IosPlatformAccess.cs]: Yep, still works in Android projects.

File: Src/packages/repositories.config (hunk 1 above)
Comment (lrdcasimir) [inline Src/packages/repositories.config]: If we're using nuget dependencies and package restore in all the SDK projects we should be able to eliminate this file entirely. It's a nuget artifact that gets created at build time it seems.

Comment (lrdcasimir) [inline Src/packages/repositories.config]: Yeah, I whacked the file and all the packages still restore and build. I'm going to delete/gitignore this then.

**Comment by lrdcasimir** [comment]
I'm going to merge this into develop if no one objects since it's blocking a customer.

**Pull request merged**